ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -0,0 +1,279 @@
+
+type ForcedMarch = 'forward' | 'backward' | 'left' | 'right';
+const effectIdToForcedMarchDir: { [id: string]: ForcedMarch } = {
+  871: 'forward', // Forward March
+  872: 'backward', // About Face
+  873: 'left', // Left Face
+  874: 'right', // Right Face
+};
+
+// For Yehehe's directional debuffs and attacks, use 0-3 for relative direction
+// e.g. 0=front, 1=right, 2=back,3=left
+// Then use that array index to map to the safe dir for that cleave:
+const yeheheCleaveToSafe = ['back', 'left', 'front', 'right'] as const;
+type YeheheDir = typeof yeheheCleaveToSafe[number] | 'unknown';
+
+const yeheheOutputStrings = {
+  front: Outputs.front,
+  back: Outputs.back,
+  right: Outputs.right,
+  left: Outputs.left,
+  unknown: Outputs.unknown,
+  next: Outputs.next,
+} as const;
+
+export interface Data extends RaidbossData {
+  yeheheTurnBuffs: number[];
+  yeheheSecondSafeDir?: YeheheDir;
+}
+
+const triggerSet: TriggerSet<Data> = {
+  id: 'Shaaloani',
+  zoneId: ZoneId.Shaaloani,
+  initData: () => ({
+    yeheheTurnBuffs: [],
+  }),
+  triggers: [
+    // ****** A-RANK: Keheniheyamewi ****** //
+    {
+      id: 'Hunt Keheni Scatterscourge',
+      type: 'StartsUsing',
+      netRegex: { id: '9B7F', source: 'Keheniheyamewi', capture: false },
+      response: Responses.getIn(),
+    },
+    {
+      id: 'Hunt Keheni Body Press',
+      type: 'StartsUsing',
+      // Unclear why there are two ids for this, but both are used.
+      netRegex: { id: ['9C7F', '96FB'], source: 'Keheniheyamewi', capture: false },
+      response: Responses.outOfMelee('alert'),
+    },
+    {
+      id: 'Hunt Keheni Slippery Scatterscourge',
+      type: 'StartsUsing',
+      netRegex: { id: '96F8', source: 'Keheniheyamewi', capture: false },
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Follow dash (in after)',
+        },
+      },
+    },
+    // Note: Because Forced March overlaps with two abilities that already have triggers,
+    // and because it applies right before the second resolves, it gets messy to combine
+    // the callouts. This would really benefit from a countdown on the Forced March info trigger.
+    {
+      id: 'Hunt Keheni Forced March Early',
+      type: 'GainsEffect',
+      netRegex: { effectId: Object.keys(effectIdToForcedMarchDir), source: 'Keheniheyamewi' },
+      condition: Conditions.targetIsYou(),
+      infoText: (_data, matches, output) => {
+        const dir = effectIdToForcedMarchDir[matches.effectId];
+        if (dir !== undefined)
+          return output[dir]!();
+      },
+      outputStrings: {
+        forward: {
+          en: 'Forced March: Forward (later)',
+        },
+        backward: {
+          en: 'Forced March: Backward (later)',
+        },
+        left: {
+          en: 'Forced March: Left (later)',
+        },
+        right: {
+          en: 'Forced March: Right (later)',
+        },
+      },
```

Comment:
unrelated to review: Forced March is a somewhat common mechanic, I wonder if we should add it to `outputs.ts`

Reply:
> unrelated to review: Forced March is a somewhat common mechanic, I wonder if we should add it to `outputs.ts`

Not opposed at all.  But it seems that usually Forced March is combined with another mech (like chariot/dynamo), so in most cases, I think we'd want the trigger output for the mech to have a combined call, whereas a call specific to just Forced March would likely be something like this (flagging that is has been applied for later, plus maybe a countdown).